BEIMA.Backend.Test/AuthenticationService/AuthenticationServiceTest.cs

```diff
@@ -0,0 +1,157 @@
+            Assert.That(authSerivce1, Is.EqualTo(authSerivce5));
+        }
+
+        [Test]
+        public void HttpRequestNoHeaders_NoClaimsReturned()
+        {
+            var authService = AuthenticationService.Instance;
+            var request = CreateHttpRequest(RequestMethod.GET);
+
+            var claims = authService.ParseToken(request);
+            Assert.That(claims, Is.Null);
+        }
+
+        [Test]
+        public void MultiPartHttpRequestNoHeaders_NoClaimsReturned()
+        {
+            var authService = AuthenticationService.Instance;
+            var request = CreateMultiPartHttpRequest("");
+
+            var claims = authService.ParseToken(request);
+            Assert.That(claims, Is.Null);
+        }
+
+        [Test]
+        public void HttpRequestWithHeaders_ClaimsReturned()
+        {
+            var authService = AuthenticationService.Instance;
+            User user = new User()
+            {
+                Username = "username",
+                Role = "role"
+            };
+
+            var token = authService.CreateToken(user);            
+            var request = CreateHttpRequest(RequestMethod.GET, authToken:token);
+
+            var claims = authService.ParseToken(request);
+            Assert.That(claims, Is.Not.Null);
+            Assert.That(claims.Username, Is.EqualTo(user.Username));
+            Assert.That(claims.Role, Is.EqualTo(user.Role));
+            Assert.That(claims.Sub, Is.EqualTo(Claims.Subject));
+            Assert.That(claims.Iss, Is.EqualTo(Claims.Issuer));
+        }
+
+        [Test]
+        public void MultiPartHttpRequestWithHeaders_ClaimsReturned()
+        {
+            var authService = AuthenticationService.Instance;
+            User user = new User()
+            {
+                Username = "username",
+                Role = "role"
+            };
+
+            var token = authService.CreateToken(user);
+            var request = CreateMultiPartHttpRequest("", authToken: token);
+
+            var claims = authService.ParseToken(request);
+            Assert.That(claims, Is.Not.Null);
+            Assert.That(claims.Username, Is.EqualTo(user.Username));
+            Assert.That(claims.Role, Is.EqualTo(user.Role));
+            Assert.That(claims.Sub, Is.EqualTo(Claims.Subject));
+            Assert.That(claims.Iss, Is.EqualTo(Claims.Issuer));
+        }
+
+        [Test]
+        public void HttpRequestWithHeader_UnsignedToken_NoClaimsReturned()
+        {
+            var authService = AuthenticationService.Instance;
+            User user = new User()
+            {
+                Username = "username",
+                Role = "role"
+            };
+
+            var token = authService.CreateToken(user);
+            var requestOne = CreateHttpRequest(RequestMethod.GET, authToken: token);
+
+            var claims = authService.ParseToken(requestOne);
+            claims.Role = "admin";
```

Comment:
I think it would be appropriate to use some Assume.That()'s here on the claims variable.